src/nvim/api/vim.c

```diff
@@ -1916,6 +1917,31 @@ Object nvim_get_proc(Integer pid, Error *err)
   return rvobj;
 }
 
+/// Selects an item in the completion popupmenu
+///
+/// When insert completion is not active, this API call is silently ignored.
+/// It is mostly useful for an external UI using |ui-popupmenu| for instance
+/// to control the popupmenu with the mouse. But it can also be used in an
+/// insert mode mapping, use <cmd> mapping |:map-cmd| to ensure the mapping
+/// doesn't end completion mode.
+///
+/// @param item   Index of the item to select, starting with zero. Pass in "-1"
+///               to select no item (restore original text).
+/// @param insert Whether the selection should be reflected in the buffer text.
+/// @param finish If true, completion will be finished with this item, and the
+///               popupmenu dissmissed. implies `insert`.
+void nvim_select_popupmenu_item(Integer item, Boolean insert, Boolean finish,
+                                Error *error)
```

Comment:
should we bother to add a `opts` parameter for future extensions?

Reply:
let's do that.